docs/calendar.md

```diff
@@ -36,6 +36,46 @@ const calendar = new Intl.DateTimeFormat().resolvedOptions().calendar;
 date.withCalendar(calendar).add({ months: 1 });
 ```
 
+### Invariants Across Calendars
+
+The following "invariants" (statements that are always true) hold for all built-in calendars, and SHOULD hold for properly-authored custom calendars:
+
+- `year`, `month`, `day` are always integers that increase as time goes forward
+- `month` and `day` are always positive integers
+- `date.month === 1` during the first month of any year
+- `date.month === date.monthsInYear` during the last month of any year
+- `month` is always continuous (no gaps)
+- Any date can be serialized to an object using only four properties: `{ year, month, day, calendar }`
+
+### Writing Cross-Calendar Code
+
+Here are best practices for writing code that will work regardless of the calendar used:
+
+- Validate or coerce the calendar of all external input. If your code receives date data from an external source, you should validate that its calendar is what you expect or coerce it to a known calendar via the `withCalendar` method. Otherwise, a sophisticated attacker could change the behavior of your app or introduce security or performance issues by introducing an unexpected calendar.
+- Use `compare` methods (e.g. `Temporal.PlainDate.compare(date1, '2000-01-01')`) instead of manually comparing individual properties (e.g. `date.year > 2000`) whose meaning may vary across calendars.
+- Never compare field values in different calendars. A `month` or `year` in one calendar is unrelated to the same property values in another calendar. If dates in different calendars must be compared, use `compare`.
+- When comparing dates for equality that might be in different calendars, convert them both to the same calendar using `withCalendar`. The same ISO date in different calendars will return `false` from the `equals` method and will return a non-zero value from `compare` because the calendars are not equal.
+- When looping through all months in a year, use `monthsPerYear` as the upper bound instead of assuming that every year has 12 months.
```

Comment:
```suggestion
- When looping through all months in a year, use `monthsInYear` as the upper bound instead of assuming that every year has 12 months.
```